### PR TITLE
Feature/df 1805 requirements voraussetzung

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

**Comment by emina-str** [comment]
when you save the ChFiche with some Requirement, it doesn't "stick", so basically I get the success message, but when I go to detail page of the same ChFiche, the Requirement is missing.

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html [hunk 1]
```diff
@@ -96,4 +97,25 @@
                         (actionClick)="handleCompetenceSetActionClick($event, competenceType, competenceSet)">
     </alv-competence-set>
   </alv-competence-items-collapse-panel>
+  <alv-competence-items-collapse-panel (toggleClick)="toggleRequirements($event)"
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html]: getting this error:
image

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/requirement-search-modal/requirement-search-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+@Component({
+  selector: 'alv-requirement-search-modal',
+  templateUrl: './requirement-search-modal.component.html',
+  styleUrls: ['./requirement-search-modal.component.scss']
+})
+export class RequirementSearchModalComponent implements OnInit {
+
+  @Input() existingOccupations: string[];
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/requirement-search-modal/requirement-search-modal.component.ts]: do we need this?

File: alv-portal-ui/src/app/competence-catalog/requirements/requirement-delete/requirement-delete.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                                 *ngFor="let chFiche of chFiches"
+                                 [isItemClickable]="true"
+                                 [multiLanguageTitle]="chFiche.title"
+                                 [superTitle]="chFiche.id"
+                                 [type]="'portal.competence-catalog.competence-sets.overview.competence-set-panel-header' | translate">
+      </alv-interactive-list-item>
+    </ng-container>
+    <ng-template #noReferences>
+        <span [attr.data-test]="'no-backlinks-placeholder'"
+              [innerHTML]="'portal.competence-catalog.requirements.delete-modal.confirmation' | translate:{competenceElementId: requirement}"></span>
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/requirements/requirement-delete/requirement-delete.component.html]: should be "requirementId" in translation parameter

File: alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  modalTitle: string;
+
+  isEdit = false;
+
+  publishedRadioButtonOptions$ = of(publishedRadioButtonOptions);
+
+  draftRadioButtonOptions$ = of(draftRadioButtonOptions);
+
+  constructor(private fb: FormBuilder,
+              private competenceElementRepository: RequirementRepository,
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts]: competenceElementRepository ?

File: alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      this.createElement();
+    }
+  }
+
+
+  cancel() {
+    this.modal.dismiss();
+  }
+
+  private updateElement() {
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts]: method name

File: alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    this.competenceElementRepository.update(this.requirement.id, {
+      description: this.form.get('description').value,
+      draft: this.form.get('draft').value,
+      published: this.form.get('published').value
+    })
+      .subscribe(this.handleSuccess.bind(this));
+  }
+
+  private createElement() {
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/requirement-modal/requirement-modal.component.ts]: method name

File: alv-portal-ui/src/app/competence-catalog/requirements/requirements-overview/requirements-overview.component.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+  ngOnInit() {
+    super.ngOnInit();
+    this.actions$ = this.isCompetenceCatalogEditor$.pipe(
+      map(isEditor => isEditor ? [this.backlinkRequirementAction, this.deleteRequirementAction] : [this.backlinkRequirementAction])
+    );
+  }
+
+  openCreateModal() {
+    const modalRef = this.modalService.openMedium(RequirementModalComponent, true);
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/requirements/requirements-overview/requirements-overview.component.ts]: shouldn't be Create and Update modals also Large?

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/requirement-search-modal/requirement-search-modal.component.ts (hunk 2 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/ch-fiches/requirement-search-modal/requirement-search-modal.component.ts]: no, thanks

File: alv-portal-ui/src/app/competence-catalog/requirements/requirements-overview/requirements-overview.component.ts (hunk 7 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/requirements/requirements-overview/requirements-overview.component.ts]: will ask Ingo

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html (hunk 1 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html]: try now please

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html [hunk 8]
```diff
@@ -1,9 +1,12 @@
 <ng-container *ngIf="{
     competenceSetsActions: competenceSetsActions$ | async,
     chFicheDescriptionActions: chFicheDescriptionActions$ | async,
-    isCompetenceCatalogEditor: isCompetenceCatalogEditor$ | async
+    isCompetenceCatalogEditor: isCompetenceCatalogEditor$ | async,
+    prerequisiteActions: prerequisiteActions$ | async
   } as observables">
   <!--   Title-->
+  <pre>{{chFiche|json}}</pre>
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html]: remove this

File: alv-portal-ui/mock-tools/db.json [hunk 9]
```diff
@@ -0,0 +1,330 @@
+{
+  "work-requirements": [
```
Comment (emina-str) [inline alv-portal-ui/mock-tools/db.json]: please then change also the names in the mock data, if it's now Prerequisite

File: alv-portal-ui/mock-tools/routes.json [hunk 10]
```diff
@@ -0,0 +1,7 @@
+{
+  "/competencecatalogservice-editor/api/work-requirements/_search*": "/_search$1",
```
Comment (emina-str) [inline alv-portal-ui/mock-tools/routes.json]: prerequisite

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche-detail/ch-fiche-detail.component.ts [hunk 11]
```diff
@@ -108,7 +108,8 @@ export class ChFicheDetailComponent extends CompetenceCatalogEditorAwareComponen
       competences: this.chFiche.competences,
       occupations: this.chFiche.occupations,
       draft: this.form.get('draft').value,
-      published: this.form.get('published').value
+      published: this.form.get('published').value,
+      prerequisiteIds: []//fixme
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche-detail/ch-fiche-detail.component.ts]: why fixme?

File: alv-portal-ui/src/app/shared/backend-services/competence-catalog/prerequisite/prerequisite-repository.service.ts [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  Prerequisite,
+  UpdatePrerequisite
+} from './prerequisite.types';
+import { Page } from '../../shared.types';
+import { SearchService } from '../search-service';
+
+@Injectable({ providedIn: 'root' })
+export class PrerequisiteRepository implements SearchService<Prerequisite> {
+
+  private readonly resourceUrl = '/competencecatalogservice-editor/api/prerequisites/';
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/competence-catalog/prerequisite/prerequisite-repository.service.ts]: NOT OK

**Comment by emina-str** [comment]
translations not ok.
translations

File: alv-portal-ui/translations.csv [hunk 13]
```diff
@@ -1264,6 +1266,19 @@ portal.competence-catalog.ch-fiches.edit-description-success-notification,Beschr
 portal.competence-catalog.ch-fiches.remove-description-success-notification,Beschreibung der CH Fiche wurde gelöscht,CH Fiche description has been deleted,La description de la fiche CH a été supprimée,La descrizione della Fiche CH è stata cancellata,
 portal.competence-catalog.ch-fiches.remove-description-dialog.title,Beschreibung löschen?,Delete description?,Supprimer la description?,Cancellare la descrizione?,
 portal.competence-catalog.ch-fiches.remove-description-dialog.confirmation-question,Möchten Sie die beschreibung löschen?,Do you really want to delete the description?,Voulez-vous vraiment supprimer la description de CH Fiche?,Vuoi davvero cancellare questa descrizione?,
+portal.competence-catalog.prerequisites.overview.title,Voraussetzungen,Prerequisites,Conditions,Requisiti,
```
Comment (emina-str) [inline alv-portal-ui/translations.csv]: there is a lot of missing translations in .csv file, but you have an label-name in htmls .. please add them to translations.csv file and if present, also the translation itself.

File: alv-portal-ui/src/app/job-advertisement/shared/job-content/job-content.component.html [hunk 14]
```diff
@@ -94,7 +94,7 @@ <h2 class="job-title mt-1 mb-4">
     </alv-collapse-panel>
 
     <alv-collapse-panel *ngIf="jobDetailModel.firstOccupation as occupation"
-                        [panelId]="jobDetailPanelId.JOB_AD_REQUIREMENTS"
+                        [panelId]="jobDetailPanelId.JOB_AD_PREREQUISITES"
```
Comment (emina-str) [inline alv-portal-ui/src/app/job-advertisement/shared/job-content/job-content.component.html]: revert

File: alv-portal-ui/src/app/job-advertisement/shared/job-content/job-content.component.ts [hunk 15]
```diff
@@ -5,7 +5,7 @@ import { JobAdvertisementUtils } from '../../../shared/backend-services/job-adve
 
 enum JobDetailPanelId {
   JOB_AD_INFO = 'job-ad-info',
-  JOB_AD_REQUIREMENTS = 'job-ad-requirements',
+  JOB_AD_PREREQUISITES = 'job-ad-prerequisites',
```
Comment (emina-str) [inline alv-portal-ui/src/app/job-advertisement/shared/job-content/job-content.component.ts]: revert

**Pull request merged**